lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -433,16 +470,51 @@ StackAllocationPromoter::promoteAllocationInBlock(SILBasicBlock *BB) {
       if (SI->getDest() != ASI)
         continue;
 
-      // The stored value is the new running value.
-      RunningVal = SI->getSrc();
+      // Special handling of entry block
+      // If we have a store [assign] in the first block, OSSA guarantees we can
+      // find the previous value stored in the stack location in RunningVal.
+      // Create destroy_value of the RunningVal.
+      // For all other blocks we may not know the previous value stored in the
+      // stack location. So we will create destroy_value in
+      // StackAllocationPromoter::fixBranchesAndUses, by getting the live-in
+      // value to the block.
+      if (BB->isEntry()) {
```

Comment:
Yes this function does handle multiple block uses. I don't think `ValueLifetimeAnalysis` is needed to figure out destroys here, because store [assign] is the point where the previous value stored in the stack location is destroyed.

Reply:
Hmm, and the stack allocation is guaranteed to be removed at some point. So, you're right, this is fine.